test/ejabberd_tests/tests/rest_SUITE.erl

```diff
@@ -0,0 +1,225 @@
+%% limitations under the License.
+%%==============================================================================
+
+-module(rest_SUITE).
+-compile(export_all).
+
+-include_lib("escalus/include/escalus.hrl").
+-include_lib("common_test/include/ct.hrl").
+
+-import(rest_helper,
+        [
+            assert_inlist/2,
+            assert_notinlist/2,
+            decode_maplist/1,
+            gett/1,
+            post/2,
+            putt/2,
+            delete/1,
+            gett/2,
+            post/3,
+            putt/3,
+            delete/2
+        ]
+    ).
+
+-define(PRT(X, Y), ct:pal("~p: ~p", [X, Y])).
+-define(OK, {<<"200">>, <<"OK">>}).
+-define(CREATED, {<<"201">>, <<"Created">>}).
+-define(ERROR, {<<"500">>, _}).
+-define(NOT_FOUND, {<<"404">>, _}).
+
+%%--------------------------------------------------------------------
+%% Suite configuration
+%%--------------------------------------------------------------------
+
+-define(REGISTRATION_TIMEOUT, 2).  %% seconds
+-define(ATOMS, [name, desc, category, action, security_policy, args, result, sender]).
+
+all() ->
+    [
+        {group, adminside},
+        {group, userside}
+    ].
+
+groups() ->
+    [{adminside, [sequence], test_cases()},
+     {userside, [sequence], user_test_cases()}
+    ].
+
+test_cases() ->
+    [assertions,
+     basic,
+     sessions,
+     messages,
+     changepassword].
+
+user_test_cases() ->
+    [user_messages].
+
+
+suite() ->
+    escalus:suite().
+
+%%--------------------------------------------------------------------
+%% Init & teardown
+%%--------------------------------------------------------------------
+
+init_per_suite(Config) ->
+    escalus:init_per_suite(Config).
+
+end_per_suite(Config) ->
+    escalus:end_per_suite(Config).
+
+init_per_group(_GroupName, Config) ->
+    escalus:create_users(Config, escalus:get_users([alice, bob])),
+    escalus:delete_users(Config, escalus:get_users([mike])).
+
+
+end_per_group(_GroupName, Config) ->
+    escalus:delete_users(Config, escalus:get_users([bob, mike])).
+
+init_per_testcase(CaseName, Config) ->
+    escalus:init_per_testcase(CaseName, Config).
+
+end_per_testcase(CaseName, Config) ->
+    escalus:end_per_testcase(CaseName, Config).
+
+%%--------------------------------------------------------------------
+%% Message tests
+%%--------------------------------------------------------------------
+
+assertions(_Config) ->
+    Lst = [<<"a">>, <<"b">>, <<"c">>],
+    assert_inlist(<<"a">>, Lst),
+    assert_inlist(<<"b">>, Lst),
+    assert_inlist(<<"c">>, Lst),
+    assert_notinlist(<<"d">>, Lst),
+    Maplst = [#{a => 1, b => 2}, #{a => 1, b => 3}, #{a => 4, b => 5}],
+    assert_inlist(#{a => 1}, Maplst),
+    assert_inlist(#{a => 1, b => 2}, Maplst),
+    assert_inlist(#{a => 4}, Maplst),
+    assert_notinlist(#{a => 5}, Maplst),
+    assert_notinlist(#{a => 1, b => 1}, Maplst).
+
+
+basic(_Config) ->
+    % list commands
+    {?OK, Lcmds} = gett(<<"/list">>),
```

Comment:
IMO, `list` is not a good name for a resource. I know its verb and noun but to me the first thought is verb and resources should be nouns. Better name for me here would be: `commands`.